# Introduction


This **[nextflow](https://www.nextflow.io/)** pipeline allows you to run several programs using a one-line command. It wraps programs to assemble, annotate, taxonomically identify, genotypic characterize and perform simple comparative genomics on prokaryotic sequencing data. This pipeline is self-contained and is NOT meant to be run in modules (not independent parts of it). This with the intention of simplifying the user interaction and the user knowledge about the bioinformatics behind this processes. However, making yourself familiar with the programs behind the scenes helps you to make informed decisions. Please go to [resources](resources.html) to read more about what each program is doing.  

## Overview of the pipeline

* **Assembly, Annotation and Classification**
  * Quality control of reads (short, long, or both)
  * Cleaning reads
  * De-novo Assembly (short, long or hybrid)
  *	Quality Control of the assembly and detection of contamination
  *	Identification of the species in the sample
  *	Identification of plasmids, phages and prophages 
  *	Genes Annotation
* **Comparative genomics (when genome reference provided)** 
    * Coverage of reference genome
    * Identification of core genome
    * Identification of Single Nucleotide Variants (SNV)
    * Identification of rearrangements and larger deletions
* **Characterization of genes of interest (when genes sequences provided)**
    * Alignments to genes of interest
    * Summary table to identy presence, absence and truncated genes
    * Generation peptides sequences from the denovo assembly




# Getting Started

## Make sure you have the required dependencies

Before you get started with the pipeline, make sure you have installed all necessary dependencies for the program to run. The ProkGenomics pipeline allows you to run several tools without the need of installing each tool by yourself. It uses **[conda](https://conda.io/projects/conda/en/latest/user-guide/getting-started.html)**, **[docker](https://www.docker.com/)**, **[singularity](https://docs.sylabs.io/guides/3.5/user-guide/introduction.html)** or **[Apptainer](https://apptainer.org/get-started/)** to create environments or containers with the necessary dependencies and tools. At the moment, the option of running this pipeline with conda is susceptible to break in some conda versions or conda with customised channels in its configuration. Therefore the ***best option for now is to run it with singularity***.

To install dependencies the easiest manner is to use **[conda](https://conda.io/projects/conda/en/latest/user-guide/getting-started.html)**. If you are using a **cluster** there are high chances that **[conda](https://conda.io/projects/conda/en/latest/user-guide/getting-started.html)** is a module already installed. Please call the module. Type the following command and use `tab` after the word `conda` to see if you have different versions available. Versions successfully tested are: `conda 4.14.0`, `conda 24.1.2`.

```
# if you are in a cluster
module add anaconda
# or
module add miniconda

```

If you have **[conda](https://conda.io/projects/conda/en/latest/user-guide/getting-started.html)** installed in your system, either in your **computer** or your **server**, you can type `conda --version` to verify you have the right version to work with. 


### Create a conda environment to run ProkGenomics

Create a **[conda](https://conda.io/projects/conda/en/latest/user-guide/getting-started.html)** environment with the required dependencies. Use the following commands. Versions successfully tested are: `nextflow 21.10.6`, `singularity-ce version 3.9.9-focal`, `singularity version 3.8.6`


```
# if you can create conda env, this could work for a cluster, server or your own computer. 
conda create -n ProkGenomics_ENV
conda activate ProkGenomics_ENV
conda install -c bioconda nextflow=21.10.6
conda install conda-forge::singularity
conda install apptainer

```

Use `conda activate ProkGenomics_ENV` every time you want to use nextflow, and the singularity profile


Alternatively, you can install **nextflow** and **singularity** without using **conda**. 

 * Install nextflow using the [nextflow installation instructions](https://www.nextflow.io/docs/latest/getstarted.html).

 * Install **singularity** following these [instructions](https://docs.sylabs.io/guides/3.0/user-guide/installation.html#overview), make sure you are following the right instructions for your system ([Linux](https://docs.sylabs.io/guides/3.0/user-guide/installation.html#install-on-linux), [Windows or Mac](https://docs.sylabs.io/guides/3.0/user-guide/installation.html#install-on-windows-or-mac))


## Download the ProkGenomics

Run the following command in your terminal

```
# when private
git clone git@github.com:Grinter-Lab/ProkGenomics.git

# when public
git clone https://github.com/Grinter-Lab/ProkGenomics.git

```

If you have issues consult the [troubleshoot section](troubleshooting.html) for help

A successful download of the repository should look like this:

```
Cloning into 'ProkGenomics'...
remote: Enumerating objects: 3, done.
remote: Counting objects: 100% (3/3), done.
remote: Compressing objects: 100% (2/2), done.
remote: Total 3 (delta 0), reused 0 (delta 0), pack-reused 0
Receiving objects: 100% (3/3), done.
```

### Execute ProkGenomics from any path

To be able to run the program from any location without using the complete path, run the following commands

1. Find out the path where your program is located
```
# move to the program folder
cd ProkGenomics/

# print your working directory
pwd 

# This will print your location something like: /path/to/dir/program/

```

2. add this path to the default executable folder. 


```
# IF you are NOT using a conda environment 
# export the path of the folder into your PATH if your install singularity and nextflow in bin
export PATH="$PATH:/path/to/dir/program/"

# IF you are working with a conda env

#find your conda env path and add the content of 
conda env list

cd /path/to/conda/env/bin
for name in /path/to/dir/program/ProkGenomics/*; 
do
  ln -s name
done
```

# Set up your project

Create a working directory for your project

```
mkdir Project1
cd Project1

```
Create a folder for your raw data with in that working directory

```
mkdir rawdata
cd rawdata
# copy your rawdata for that project inside this folder
```

Please run one ProkGenomics pipeline per working directory. You can run multiple ProkGenomics at a time if the working directories are different. Several runs in the same working directory cause conflict with the *.nextflow.logs* and your pipeline will fail.  



##  How to transfer raw data to cluster/server

Open a tab in your terminal from your local computer

```
# from tab in your local computer
# scp <location in the server, notice structue as serve:path> <location in your computer where you want the file to go>
# notice that the wild card allows you to move all files ending in fq.gz. if your files have a different extention please modify accordently (ex. fastq.gz)

scp *fq.gz <username>@<cluster_name>:/srv/home/username/folder/

```

