# Write Methods

<img src="images/Method.png" width="620">

 1. Input short reads paired ends from whole genome bacteria sequencing. 
 2. [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) is used to determined quality control of rawdata.
 3. [Trimmomatic](https://github.com/timflutre/trimmomatic) is used for trimming adaptor in the sequencing data.
 4. [Unicycler](https://github.com/rrwick/Unicycler) is used to perform de-novo assembly.
 5. [Plasclass](https://github.com/Shamir-Lab/PlasClass) predicts plasmids in the assembled sequence.
 6. [CheckV](https://bitbucket.org/berkeleylab/checkv/src/master/) predicts phages in the assembled sequence.
 7. [CheckM](https://github.com/Ecogenomics/CheckM) assess the assembly quality, contamination and completeness.
 8. bash script to split the chromosome, plasmid, and phage contigs in different files.
 9. [Pharokka](https://github.com/gbouras13/pharokka) predicts genes and annotation in  phage contigs.
 10. [Prokka](https://github.com/tseemann/prokka) predicts genes and annotation in  plasmid contigs.
 11. [Prokka](https://github.com/tseemann/prokka) predicts genes and annotation in  chromosome contigs.
 12. [GTDB](https://github.com/Ecogenomics/GTDBTk) classification of chromosome contig
 13. When genes of interest are input [Prodigal](https://github.com/hyattpd/Prodigal) is used to predict genes in the assemblies and [Blast](https://blast.ncbi.nlm.nih.gov/Blast.cgi) to algin them to genes of interest. [fromAssembly2Feature](https://github.com/LPerlaza/fromAssembly2Feature) curates alignments and summary tables of the genes of interest.
 14. Input reference genome.
 15. [Minimap2](https://github.com/lh3/minimap2) is used to map the reads to the reference genome.
 16. [Samtools](https://github.com/samtools/samtools) and [Qualimap](http://qualimap.conesalab.org/) are used to summarise mapping and coverage
 17. [Snippy](https://github.com/tseemann/snippy) identifies Single Nucleotide Variant (SNV) using the raw reads and the raw reads.
 
 
## Programs description and citations
 
|Tool|Description|Paper|
| :---|:---|:---|
|[FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/)|Sequence quality controls|Andrews S. (2010). FastQC: a quality control tool for high throughput sequence data. Available online at: http://www.bioinformatics.babraham.ac.uk/projects/fastqc|
|[Trimmomatic](https://github.com/timflutre/trimmomatic)|Trim primer adaptor from reads|Bolger AM, Lohse M, Usadel B. Trimmomatic: a flexible trimmer for Illumina sequence data. Bioinformatics. 2014 Aug 1;30(15):2114-20. doi: 10.1093/bioinformatics/btu170. Epub 2014 Apr 1. PMID: 24695404; PMCID: PMC4103590. https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4103590/|
|[Unicycler](https://github.com/rrwick/Unicycler)|De novo assembly|Wick RR, Judd LM, Gorrie CL, Holt KE (2017) Unicycler: Resolving bacterial genome assemblies from short and long sequencing reads. PLOS Computational Biology 13(6): e1005595. https://doi.org/10.1371/journal.pcbi.1005595|
|[Plasclass](https://github.com/Shamir-Lab/PlasClass)|Plasmids prediction|Pellow D, Mizrahi I, Shamir R (2020) PlasClass improves plasmid sequence classification. PLOS Computational Biology 16(4): e1007781. https://doi.org/10.1371/journal.pcbi.1007781|
|[CheckV](https://bitbucket.org/berkeleylab/checkv/src/master/)|Phage prediction|Nayfach, S., Camargo, A.P., Schulz, F. et al. CheckV assesses the quality and completeness of metagenome-assembled viral genomes. Nat Biotechnol 39, 578–585 (2021). https://doi.org/10.1038/s41587-020-00774-7|
|[CheckM](https://github.com/Ecogenomics/CheckM)|Assembly quality controls|Parks DH, Imelfort M, Skennerton CT, Hugenholtz P, Tyson GW. CheckM: assessing the quality of microbial genomes recovered from isolates, single cells, and metagenomes. Genome Res. 2015 Jul;25(7):1043-55. doi: 10.1101/gr.186072.114. Epub 2015 May 14. PMID: 25977477; PMCID: PMC4484387. https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4484387/|
|[Prokka](https://github.com/tseemann/prokka)|Prokaryotic gene annotation|Torsten Seemann, Prokka: rapid prokaryotic genome annotation, Bioinformatics, Volume 30, Issue 14, July 2014, Pages 2068–2069, https://doi.org/10.1093/bioinformatics/btu153|
|[Pharokka](https://github.com/gbouras13/pharokka)|Phage gene annotation|Bouras G, Nepal R, Houtak G, Psaltis AJ, Wormald PJ, Vreugde S. Pharokka: a fast scalable bacteriophage annotation tool. Bioinformatics. 2023 Jan 1;39(1):btac776. doi: 10.1093/bioinformatics/btac776. PMID: 36453861; PMCID: PMC9805569. https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9805569/|
|[GTDB](https://github.com/Ecogenomics/GTDBTk)|Taxonomic classifications |Donovan H Parks, Maria Chuvochina, Christian Rinke, Aaron J Mussig, Pierre-Alain Chaumeil, Philip Hugenholtz, GTDB: an ongoing census of bacterial and archaeal diversity through a phylogenetically consistent, rank normalized and complete genome-based taxonomy, Nucleic Acids Research, Volume 50, Issue D1, 7 January 2022, Pages D785–D794, https://doi.org/10.1093/nar/gkab776|
|[Blast](BLAST)| Basic Local Alignment Search Tool| Camacho C., Coulouris G., Avagyan V., Ma N., Papadopoulos J., Bealer K., Madden T.L. (2008) “BLAST+: architecture and applications.” BMC Bioinformatics 10:421. https://doi.org/10.1186/1471-2105-10-421 |
 [Prodigal]((https://github.com/hyattpd/Prodigal))|prediction of prokaryotic genes|Hyatt, D., Chen, GL., LoCascio, P.F. et al. Prodigal: prokaryotic gene recognition and translation initiation site identification. BMC Bioinformatics 11, 119 (2010). https://doi.org/10.1186/1471-2105-11-119 |
|[Snippy](https://github.com/tseemann/snippy)|Single Nucleotide Variant (SNV) detection|Seemann T (2015). snippy: fast bacterial variant calling from NGS reads. https://github.com/tseemann/snippy|
|[Minimap2](https://github.com/lh3/minimap2)|Sequence alignment to reference genome|Heng Li, New strategies to improve minimap2 alignment accuracy, Bioinformatics, Volume 37, Issue 23, December 2021, Pages 4572–4574, https://doi.org/10.1093/bioinformatics/btab705|
[Samtools](https://github.com/samtools/samtools)|Parsing and summarizing mapping (SAM/BAM) files|Twelve years of SAMtools and BCFtools. Petr Danecek, James K Bonfield, Jennifer Liddle, John Marshall, Valeriu Ohan, Martin O Pollard, Andrew Whitwham, Thomas Keane, Shane A McCarthy, Robert M Davies, Heng Li GigaScience, Volume 10, Issue 2, February 2021, giab008, https://doi.org/10.1093/gigascience/giab008|
|[Qualimap](http://qualimap.conesalab.org/)|Quality control of mapping| Fernando García-Alcalde, Konstantin Okonechnikov, José Carbonell, Luis M. Cruz, Stefan Götz, Sonia Tarazona, Joaquín Dopazo, Thomas F. Meyer, Ana Conesa, Qualimap: evaluating next-generation sequencing alignment data, Bioinformatics, Volume 28, Issue 20, October 2012, Pages 2678–2679, https://doi.org/10.1093/bioinformatics/bts503|
|[fromAssembly2feature](https://github.com/LPerlaza/fromAssembly2Feature)|Emulates manually curated alignmets|Perlaza-Jimenez L (2019) From Assemblies to Feature: identifying genes of interest and generating accurate alignments and genes summary. https://github.com/LPerlaza/fromAssembly2Feature|




