# Resources

**Programs description and citations**

|Tool|Description|Paper|
| :---|:---|:---|
|[FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/)|Sequence quality controls|Andrews S. (2010). FastQC: a quality control tool for high throughput sequence data. Available online at: http://www.bioinformatics.babraham.ac.uk/projects/fastqc|
|[Trimmomatic](https://github.com/timflutre/trimmomatic)|Trim primer adaptor from reads|Bolger AM, Lohse M, Usadel B. Trimmomatic: a flexible trimmer for Illumina sequence data. Bioinformatics. 2014 Aug 1;30(15):2114-20. doi: 10.1093/bioinformatics/btu170. Epub 2014 Apr 1. PMID: 24695404; PMCID: PMC4103590. https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4103590/|
|[Unicycler](https://github.com/rrwick/Unicycler)|De novo assembly|Wick RR, Judd LM, Gorrie CL, Holt KE (2017) Unicycler: Resolving bacterial genome assemblies from short and long sequencing reads. PLOS Computational Biology 13(6): e1005595. https://doi.org/10.1371/journal.pcbi.1005595|
|[Plasclass](https://github.com/Shamir-Lab/PlasClass)|Plasmids prediction|Pellow D, Mizrahi I, Shamir R (2020) PlasClass improves plasmid sequence classification. PLOS Computational Biology 16(4): e1007781. https://doi.org/10.1371/journal.pcbi.1007781|
|[CheckV](https://bitbucket.org/berkeleylab/checkv/src/master/)|Phage prediction|Nayfach, S., Camargo, A.P., Schulz, F. et al. CheckV assesses the quality and completeness of metagenome-assembled viral genomes. Nat Biotechnol 39, 578–585 (2021). https://doi.org/10.1038/s41587-020-00774-7|
|[Prokka](https://github.com/tseemann/prokka)|Prokaryotic gene annotation|Torsten Seemann, Prokka: rapid prokaryotic genome annotation, Bioinformatics, Volume 30, Issue 14, July 2014, Pages 2068–2069, https://doi.org/10.1093/bioinformatics/btu153|
|[Pharokka](https://github.com/gbouras13/pharokka)|Phage gene annotation|Bouras G, Nepal R, Houtak G, Psaltis AJ, Wormald PJ, Vreugde S. Pharokka: a fast scalable bacteriophage annotation tool. Bioinformatics. 2023 Jan 1;39(1):btac776. doi: 10.1093/bioinformatics/btac776. PMID: 36453861; PMCID: PMC9805569. https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9805569/|
|[CheckM](https://github.com/Ecogenomics/CheckM)|Assembly quality controls|Parks DH, Imelfort M, Skennerton CT, Hugenholtz P, Tyson GW. CheckM: assessing the quality of microbial genomes recovered from isolates, single cells, and metagenomes. Genome Res. 2015 Jul;25(7):1043-55. doi: 10.1101/gr.186072.114. Epub 2015 May 14. PMID: 25977477; PMCID: PMC4484387. https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4484387/|
|[Snippy](https://github.com/tseemann/snippy)|Single Nucleotide Variant (SNV) detection|

**Other programs and pipelines**


https://bactopia.github.io/v3.0.0/

https://proksee.ca/

https://genome.usegalaxy.org.au/
