<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 4 Run Pipeline | Assembly, Annotation, and Comparative Genomics for Prokaryotes</title>
<meta name="author" content="Laura Perlaza-Jiménez (PhD)">
<meta name="description" content="Remember to put the program in your $PATH if you haven’t done it. This step has to be done every time you start a new terminal session. If you want to make this change permanent you could modify...">
<meta name="generator" content="bookdown 0.37 with bs4_book()">
<meta property="og:title" content="Chapter 4 Run Pipeline | Assembly, Annotation, and Comparative Genomics for Prokaryotes">
<meta property="og:type" content="book">
<meta property="og:description" content="Remember to put the program in your $PATH if you haven’t done it. This step has to be done every time you start a new terminal session. If you want to make this change permanent you could modify...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 4 Run Pipeline | Assembly, Annotation, and Comparative Genomics for Prokaryotes">
<meta name="twitter:description" content="Remember to put the program in your $PATH if you haven’t done it. This step has to be done every time you start a new terminal session. If you want to make this change permanent you could modify...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.6.1/transition.js"></script><script src="libs/bs3compat-0.6.1/tabs.js"></script><script src="libs/bs3compat-0.6.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Assembly, Annotation, and Comparative Genomics for Prokaryotes</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> START HERE</a></li>
<li><a class="" href="introduction.html"><span class="header-section-number">2</span> Introduction</a></li>
<li><a class="" href="getting-started.html"><span class="header-section-number">3</span> Getting Started</a></li>
<li><a class="active" href="run-pipeline.html"><span class="header-section-number">4</span> Run Pipeline</a></li>
<li><a class="" href="understand-outputs-in-development.html"><span class="header-section-number">5</span> Understand Outputs (in development)</a></li>
<li><a class="" href="write-methods.html"><span class="header-section-number">6</span> Write Methods</a></li>
<li><a class="" href="troubleshooting.html"><span class="header-section-number">7</span> Troubleshooting</a></li>
<li><a class="" href="resources.html"><span class="header-section-number">8</span> Resources</a></li>
<li><a class="" href="command-line-intro.html"><span class="header-section-number">9</span> Command Line Intro</a></li>
<li><a class="" href="connect-to-the-cluster.html"><span class="header-section-number">10</span> Connect to the cluster</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/Grinter-Lab/DocumentationProkGenomics">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="run-pipeline" class="section level1" number="4">
<h1>
<span class="header-section-number">4</span> Run Pipeline<a class="anchor" aria-label="anchor" href="#run-pipeline"><i class="fas fa-link"></i></a>
</h1>
<p>Remember to put the <a href="getting-started.">program in your $PATH</a> if you haven’t done it. This step has to be done every time you start a new terminal session. If you want to make this change permanent you could modify your <a href="https://stackoverflow.com/questions/14637979/how-to-permanently-set-path-on-linux-unix">bash profile</a> (don’t play around with it if you don’t feel confident about it)</p>
<p>If you added the program to your <code>$PATH</code> successfully you should be able to run</p>
<pre><code>#remember to activate the conda env if you are using one
conda activate nextflow

#run the pipeline
ProkGenomics
</code></pre>
<p>If it starts correctly, you will see something like</p>
<pre><code>N E X T F L O W  ~  version 21.10.6
Launching `main.nf` [focused_noether] - revision: eb930f0e69
</code></pre>
<p>If you don’t see a version of that go to <a href="troubleshooting.html">troubleshoot</a> to look for possible solutions</p>
<div id="simple-run" class="section level2" number="4.1">
<h2>
<span class="header-section-number">4.1</span> Simple run<a class="anchor" aria-label="anchor" href="#simple-run"><i class="fas fa-link"></i></a>
</h2>
<pre><code>#remember to activate the conda env if you are using one, and if you haven't activated it yet
conda activate nextflow

#run the pipeline
ProkGenomics --sample_name '1-77321' -profile conda
</code></pre>
<p><strong>Parameters you can use:</strong></p>
<div class="inline-table"><table class="table table-sm">
<colgroup>
<col width="44%">
<col width="55%">
</colgroup>
<thead><tr class="header">
<th align="left">Command</th>
<th align="left">Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><code>--sample_path ./rawdata/</code></td>
<td align="left">The default path for the reads is the folder rawdata in the working directory. If you followed the instructions for <a href="getting-started.html">setting up the working folder</a>. if you have your reads somewhere else you should change this parameter to that path.</td>
</tr>
<tr class="even">
<td align="left"><code>--sample_name 1-77321</code></td>
<td align="left">The sample name is the prefix of your samples files. it doesn’t have a default because I don’t know your sample names.</td>
</tr>
<tr class="odd">
<td align="left"><code>--assembly_type short</code></td>
<td align="left">This parameter can be short long or hybrid. The default is ‘short’. if you have short reads you don’t have to specify this parameter. If you pick the argument long or hybrid the longreads parameter should be specify. For hybrid make sure to give a path for short and long reads.</td>
</tr>
<tr class="even">
<td align="left"><code>--longreads  ./rawdata/longreads/</code></td>
<td align="left">Path to the long reads.</td>
</tr>
<tr class="odd">
<td align="left"><code>--threads 16</code></td>
<td align="left">Number of threats to use. More threats faster your processing. Make sure you know what is available for you.</td>
</tr>
<tr class="even">
<td align="left"><code>--outdir 1-77321</code></td>
<td align="left">The results will be in a folder in the working directory with the same sample name and _results ex. 1-77321_results.</td>
</tr>
<tr class="odd">
<td align="left"><code>--reference reference/ReferenceGenome.fasta</code></td>
<td align="left">If you have a reference genome put the path here. This will activate all the comparative genomics steps.</td>
</tr>
<tr class="even">
<td align="left"><code>--adapter_file TruSeq3-PE.fa</code></td>
<td align="left">To trim your short reads you need to specify what adaptors where used when sequencing. Arguments are <code>TruSeq2-SE.fa</code>, <code>TruSeq2-PE.fa</code>, <code>TruSeq3-PE.fa</code>. The default is <code>TruSeq3-PE.fa</code>.</td>
</tr>
<tr class="odd">
<td align="left"><code>--genes_interest GenesBD/*fasta</code></td>
<td align="left">Path to a folder that contains all genes of interest. The correct formating is a genes per file in fasta format.</td>
</tr>
</tbody>
</table></div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="getting-started.html"><span class="header-section-number">3</span> Getting Started</a></div>
<div class="next"><a href="understand-outputs-in-development.html"><span class="header-section-number">5</span> Understand Outputs (in development)</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#run-pipeline"><span class="header-section-number">4</span> Run Pipeline</a></li>
<li><a class="nav-link" href="#simple-run"><span class="header-section-number">4.1</span> Simple run</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/Grinter-Lab/DocumentationProkGenomics/blob/master/02-RunPipeline.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/Grinter-Lab/DocumentationProkGenomics/edit/master/02-RunPipeline.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Assembly, Annotation, and Comparative Genomics for Prokaryotes</strong>" was written by Laura Perlaza-Jiménez (PhD). It was last built on Compiled: February 26, 2024.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
