<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 5 Understand Outputs (in development) | Assembly, Annotation, and Comparative Genomics for Prokaryotes</title>
<meta name="author" content="Laura Perlaza-Jiménez (PhD)">
<meta name="description" content="This pipeline produce several different outputs. The main results are stored in a folder names main_results  5.1 OUTPUTS Folder structure 1-77321_results ├── fastqc │   ├──...">
<meta name="generator" content="bookdown 0.37 with bs4_book()">
<meta property="og:title" content="Chapter 5 Understand Outputs (in development) | Assembly, Annotation, and Comparative Genomics for Prokaryotes">
<meta property="og:type" content="book">
<meta property="og:description" content="This pipeline produce several different outputs. The main results are stored in a folder names main_results  5.1 OUTPUTS Folder structure 1-77321_results ├── fastqc │   ├──...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 5 Understand Outputs (in development) | Assembly, Annotation, and Comparative Genomics for Prokaryotes">
<meta name="twitter:description" content="This pipeline produce several different outputs. The main results are stored in a folder names main_results  5.1 OUTPUTS Folder structure 1-77321_results ├── fastqc │   ├──...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.6.1/transition.js"></script><script src="libs/bs3compat-0.6.1/tabs.js"></script><script src="libs/bs3compat-0.6.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Assembly, Annotation, and Comparative Genomics for Prokaryotes</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> START HERE</a></li>
<li><a class="" href="introduction.html"><span class="header-section-number">2</span> Introduction</a></li>
<li><a class="" href="getting-started.html"><span class="header-section-number">3</span> Getting Started</a></li>
<li><a class="" href="run-pipeline.html"><span class="header-section-number">4</span> Run Pipeline</a></li>
<li><a class="active" href="understand-outputs-in-development.html"><span class="header-section-number">5</span> Understand Outputs (in development)</a></li>
<li><a class="" href="write-methods.html"><span class="header-section-number">6</span> Write Methods</a></li>
<li><a class="" href="troubleshooting.html"><span class="header-section-number">7</span> Troubleshooting</a></li>
<li><a class="" href="resources.html"><span class="header-section-number">8</span> Resources</a></li>
<li><a class="" href="command-line-intro.html"><span class="header-section-number">9</span> Command Line Intro</a></li>
<li><a class="" href="connect-to-the-cluster.html"><span class="header-section-number">10</span> Connect to the cluster</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/Grinter-Lab/DocumentationProkGenomics">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="understand-outputs-in-development" class="section level1" number="5">
<h1>
<span class="header-section-number">5</span> Understand Outputs (in development)<a class="anchor" aria-label="anchor" href="#understand-outputs-in-development"><i class="fas fa-link"></i></a>
</h1>
<p>This pipeline produce several different outputs. The main results are stored in a folder names main_results</p>
<div id="outputs" class="section level2" number="5.1">
<h2>
<span class="header-section-number">5.1</span> OUTPUTS<a class="anchor" aria-label="anchor" href="#outputs"><i class="fas fa-link"></i></a>
</h2>
<p>Folder structure</p>
<pre><code>1-77321_results
├── fastqc
│   ├── 1-77321-LFA6246_L2_1_fastqc.html
│   ├── 1-77321-LFA6246_L2_1_fastqc.zip
│   ├── 1-77321-LFA6246_L2_2_fastqc.html
│   ├── 1-77321-LFA6246_L2_2_fastqc.zip
│   ├── 1-77321-LFA6246_L2.R1.trim_fastqc.html
│   ├── 1-77321-LFA6246_L2.R1.trim_fastqc.zip
│   ├── 1-77321-LFA6246_L2.R1.unpaired.trim_fastqc.html
│   ├── 1-77321-LFA6246_L2.R1.unpaired.trim_fastqc.zip
│   ├── 1-77321-LFA6246_L2.R2.trim_fastqc.html
│   ├── 1-77321-LFA6246_L2.R2.trim_fastqc.zip
│   ├── 1-77321-LFA6246_L2.R2.unpaired.trim_fastqc.html
│   ├── 1-77321-LFA6246_L2.R2.unpaired.trim_fastqc.zip
│   └── software_details.txt
├── trimmomatic
│   ├── 1-77321-LFA6246_L2.R1.trim.fastq
│   ├── 1-77321-LFA6246_L2.R1.unpaired.trim.fastq
│   ├── 1-77321-LFA6246_L2.R2.trim.fastq
│   ├── 1-77321-LFA6246_L2.R2.unpaired.trim.fastq
│   └── software_details.txt
└── unicycler
    ├── 1-77321-LFA6246_L2
    │   ├── 001_spades_graph_k027.gfa
    │   ├── 001_spades_graph_k053.gfa
    │   ├── 001_spades_graph_k071.gfa
    │   ├── 001_spades_graph_k087.gfa
    │   ├── 001_spades_graph_k099.gfa
    │   ├── 001_spades_graph_k111.gfa
    │   ├── 001_spades_graph_k119.gfa
    │   ├── 001_spades_graph_k127.gfa
    │   ├── 002_depth_filter.gfa
    │   ├── 003_overlaps_removed.gfa
    │   ├── 004_bridges_applied.gfa
    │   ├── 005_final_clean.gfa
    │   ├── assembly.fasta &lt;&lt;&lt;&lt;&lt; Complete de novo assembly
    │   ├── assembly.gfa
    │   └── unicycler.log
    └── software_details.txt
├── checkm
│   ├── 1-77321-LFA6246_L2.tsv &lt;&lt;&lt;&lt;&lt; QC de novo assembly
│   └── software_details.txt
├── checkv 
│   ├── 1-77321-LFA6246_L2
│   │   ├── complete_genomes.tsv
│   │   ├── completeness.tsv
│   │   ├── contamination.tsv
│   │   ├── proviruses.fna
│   │   ├── quality_summary.tsv &lt;&lt;&lt;&lt;&lt; Phages or Provirus detected
│   │   └── viruses.fna
│   └── software_details.txt
├── prokka
│   ├── 1-77321-LFA6246_L2_annotation_output
│   │   ├── 1-77321-LFA6246_L2.err
│   │   ├── 1-77321-LFA6246_L2.faa
│   │   ├── 1-77321-LFA6246_L2.ffn
│   │   ├── 1-77321-LFA6246_L2.fna
│   │   ├── 1-77321-LFA6246_L2.fsa
│   │   ├── 1-77321-LFA6246_L2.gbk &lt;&lt;&lt;&lt;&lt; Gene Annotation
│   │   ├── 1-77321-LFA6246_L2.gff &lt;&lt;&lt;&lt;&lt; Gene Annotation
│   │   ├── 1-77321-LFA6246_L2.log
│   │   ├── 1-77321-LFA6246_L2.sqn
│   │   ├── 1-77321-LFA6246_L2.tbl
│   │   ├── 1-77321-LFA6246_L2.tsv
│   │   └── 1-77321-LFA6246_L2.txt
│   └── software_details.txt
├── pharokka
│   ├── 1-77321-LFA6246_L2_annotation_output
│   │   ├── 1-77321-LFA6246_L2.err
│   │   ├── 1-77321-LFA6246_L2.faa
│   │   ├── 1-77321-LFA6246_L2.ffn
│   │   ├── 1-77321-LFA6246_L2.fna
│   │   ├── 1-77321-LFA6246_L2.fsa
│   │   ├── 1-77321-LFA6246_L2.gbk
│   │   ├── 1-77321-LFA6246_L2.gff &lt;&lt;&lt;&lt;&lt; Phages or Provirus annotation
│   │   ├── 1-77321-LFA6246_L2.log
│   │   ├── 1-77321-LFA6246_L2.sqn
│   │   ├── 1-77321-LFA6246_L2.tbl
│   │   ├── 1-77321-LFA6246_L2.tsv
│   │   └── 1-77321-LFA6246_L2.txt
│   └── software_details.txt

  
</code></pre>
<div class="inline-table"><table class="table table-sm">
<thead><tr class="header">
<th align="left">File</th>
<th align="left">Description</th>
</tr></thead>
<tbody><tr class="odd">
<td align="left"></td>
<td align="left"></td>
</tr></tbody>
</table></div>
<p>Folder with each analysis will be generated.</p>
<p>folder <strong>test_output</strong>:
This folder contains all the intermedia files used for the program. These files will help you to check in detail where your alignment come from. In case you are puzzle by your final table. Each folder contains:</p>
<p><strong>GenomeName.chr.genes.faa</strong>: all predicted genes in AA<br><strong>GenomeName.chr.genes.fasta</strong>: all predicted genes in nt<br><strong>GenomeName.chr.genes.gff</strong>: all predicted genes in gff format with the fasta file at the end<br><strong>GenomeName.chromosome.GeneName.Blast.txt</strong>: blast results of the gene against the genome<br><strong>GenomeName.chromosome.GeneName.fasta.plusRef.fasta</strong>: fasta of gene reference and gene in the genome</p>
<p>folder <em>test_results</em><br>
This folder contains all final results files and a folder with the predicted peptides that match with the genes of interest<br><strong>GeneName.fasta.nt_alignment.fasta</strong>: The alignment of each gene of interest for all the genomes analysed<br><strong>test.alignments.description.txt</strong>: table with the descriptive information of the alignments, stop codons, gaps, insertions, SNPs, N.copies (numbers of copies)<br><strong>Peptides</strong> (folder): predicted peptides that match with the genes of interest</p>
<p>Additional files when running any of the –Kleb, –Esch or –Ent options</p>
<p>folder <strong>test_output</strong>:<br>
Each assembly has two folders one for the chromosome and one for the plasmid. Examples here are about the chromosome, <em>.plasmid_</em> are for plasmids<br><strong>GenomeName.chromosome_contigslist.txt</strong>:list of contigs in the assembly that are chromosomal<br><strong>GenomeName.chromosome.fasta</strong>: fasta file of chromosomal contigs<br><strong>GenomeName.fsa_nt.chromosomesummary.txt</strong>: summary results from chromosome prediction from mlplasmid</p>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="run-pipeline.html"><span class="header-section-number">4</span> Run Pipeline</a></div>
<div class="next"><a href="write-methods.html"><span class="header-section-number">6</span> Write Methods</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#understand-outputs-in-development"><span class="header-section-number">5</span> Understand Outputs (in development)</a></li>
<li><a class="nav-link" href="#outputs"><span class="header-section-number">5.1</span> OUTPUTS</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/Grinter-Lab/DocumentationProkGenomics/blob/master/03-UnderstandOutputs.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/Grinter-Lab/DocumentationProkGenomics/edit/master/03-UnderstandOutputs.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Assembly, Annotation, and Comparative Genomics for Prokaryotes</strong>" was written by Laura Perlaza-Jiménez (PhD). It was last built on Compiled: February 23, 2024.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
